### PR TITLE
feat: add a persistence service

### PR DESCRIPTION
**Pull request by chrisharrison**

### CODE REVIEW

**Comment by chrisharrison** [comment]
Thanks for the review. All great points.

For the random stuff: yeh was trying to separate each instance so they didn't overwrite each other. But didn't properly think it through or test it! Was thinking to probably add something like:

```
class LocalStoragePersistenceService {
...
constructor(appId: string) {
}
...
```

Making it an implementation detail for that service rather than adding it to the interface. Just so that different implementations can do their own management. What do you think?

I'll update this PR with your suggestions on Monday.

**Review by emibcn — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -142,6 +142,39 @@ export default withServiceWorkerUpdater(Updater, {
...
+
+  isUpdateNeeded(): boolean {
+    return false
+  }
+}
+```
+
 ## :trophy: Thanks to
 
 - [<img align="center" width="30px" height="30px" src="https://avatars.githubusercontent.com/u/2166114?s=40&v=4" />](https://github.com/aeharding) [Alexander Harding](https://github.com/aeharding) for [translating the library into Typescript](https://github.com/emibcn/service-worker-updater/pull/55).
```
[the remark this answers is not in the document]
Comment (emibcn) [reply README.md:180]: If you don't want to be added, say it and I'll merge the PR without it.

**Comment by emibcn** [comment]
Once merged, I'll bump the version in `package.json` and publish it to NPM.

**Comment by chrisharrison** [comment]
Thanks for the approval, let's merge and release!

**Comment by emibcn** [comment]
Released!

**Pull request merged**